FantasyUniverse/Services/QuestionDataService.swift

```diff
@@ -1,41 +1,52 @@
 //
-//  QuestionViewData.swift
+//  QuestionDataService.swift
 //  FantasyUniverse
 //
-//  Created by TarasPeregrinus on 23.03.2022.
+//  Created by TarasPeregrinus on 19.04.2022.
 //
 
 import Foundation
-import FirebaseFirestoreSwift
 
-enum QuestionDataService {
+class QuestionDataService {
     
-    static func formatQuestions(_ questions: [Question]) -> [QuestionViewData]? {
-        var questionViewData: [QuestionViewData]?
-        for question in questions {
-            // swiftlint:disable line_length
-            guard let questionString = question.question, let optionA = question.optionA, let optionB = question.optionB, let optionC = question.optionC, let optionD = question.optionD, let correct = question.correct
-            else {
-                if let missingValuesMessages = missingValuesMessages(of: question) {
-                    debugPrint(missingValuesMessages)
-                }
-                questionViewData = nil
-                break
+    static let shared = QuestionDataService()
+    private var cachedQuestions: CachedQuestions
+    private let firebaseRepository: FirebaseRepository
+    
+    private init() {
+        self.cachedQuestions = CachedQuestions()
+        self.firebaseRepository = FirebaseRepository()
+    }
+    
+    func refreshQuestions() {
+        self.firebaseRepository.getAllQuestions { data in
+            self.cachedQuestions.cacheObjects(data)
+        }
```

Comment:
I think this would just add more questions, maybe remove all existing from the cache first?

Reply:
@dzoanb In this case it should not add more questions to existing ones. We use : 
 func cacheElements(_ elements: [QuestionCollection]) {
        cachedElements = elements
    }
This will re-asign value to cachedElements, not append more values.